### PR TITLE
CCSMCDS-13 Map UMMC STIDES data

### PR DESCRIPTION
**Pull request by yunwwang**
This PR fixes GitHub Issue #15 

Change Log:
* Created convert-strides.mjs as standalone app to covert CVS format to JSON hash using MRN column as key.
* Match FHIR Data and Strides Data using Patient's MRN and DiagnosticReport's identifier.
* Create FHIR Procedure from Strides data
* Update DiagnosticReport's conclusionCodes from StridesData
* Create Unit Test

To Test this:
1. Save XLSX data as CSV data to .\data\strides.csv
2. Run `npm run convert-csv` to populate strides.js 
3. Run `npm run test-mocha` for unit testing

### CODE REVIEW

**Comment by mickohanlon23** [comment]
@yunwwang Is this PR ready for review? I so, would you mind un-marking as draft and rebasing when you have a moment?

File: src/hooks/useCds/convert-strides.mjs
Comment (mickohanlon23) [inline src/hooks/useCds/convert-strides.mjs:22]: Should we add more graceful handling for when a CSV file is not available in the supplied directory?

File: .gitignore
Comment (mickohanlon23) [inline .gitignore:38]: Should we add strides.js to gitignore, so that converted data is not pushed, and we keep the empty `stridesData` object as the one that is checked in to strides.js?

File: src/hooks/useCds/translate.js
Comment (mickohanlon23) [inline src/hooks/useCds/translate.js]:
```suggestion
  const mrn = patientDataMap.Patient[0].identifier?.find(id => id.type?.text === 'MRN')?.value;
```
Should we add graceful handling for when a Patient.identifier does not exist?

File: package.json
Comment (mickohanlon23) [inline package.json:113]: It may be nice to add a very brief README description describing how to run `convert-csv`, and what it does.

**Review by mickohanlon23 — CHANGES_REQUESTED** [review]
Nice! I reviewed the code and tested by converting the mock UMMC Strides data Excel file to CSV, and bringing it into data/strides.csv. The logic makes sense - it appears that you properly add a conclusion code to a matching diagnostic report, and add a procedure if one is indicated by the STRIDES data. I was able to successfully create the stridesData object using `npm run convert-csv` I made some minor suggestions but overall this looks like it is ready to test with the real STRIDES data!

File: test/translate-test.js
Comment (mickohanlon23) [inline test/translate-test.js]:
```suggestion
/* eslint-disable no-unused-expressions */
const chai = require('chai');
```
Consider adding this to the translate-test.js file, since I am seeing the following esline warning throughout the `expect` statements:

> Expected an assignment or function call and instead saw an expression. eslint[no-unused-expressions](https://eslint.org/docs/rules/no-unused-expressions)

File: .gitignore
Comment (yunwwang) [reply .gitignore:38]: The challenge is that the JavaScript module bundler (like Webpack) used in build process attempts to resolve all import statements at compile time. If the file doesn't exist at that point, it results in a compilation error, even though the dynamic import is designed to handle such cases at runtime.

File: README.md
Comment (mickohanlon23) [inline README.md]:
```suggestion
You can run this following command to convert CSV data to JSON object. You need to run this if the CSV file is changed:
```

File: README.md
Comment (mickohanlon23) [inline README.md]:
```suggestion
The `[filepath]` is the location/path to the CSV file. Example:
```

File: src/hooks/useCds/convert-strides.mjs
Comment (mickohanlon23) [inline src/hooks/useCds/convert-strides.mjs]:
```suggestion
console.log("CSV file provided is converted to JSON successfully.")
```

**Review by mickohanlon23 — APPROVED** [review]
I left a couple comments to fix word spellings. This PR also still needs to be rebased / merged with the pilot branch. Approving pending those two updates, since I don't anticipate needing to review further after those updates are made. Nice job!

File: .gitignore
Comment (mickohanlon23) [reply .gitignore:38]: This was addressed by the most recent group of commits, through a prebuild script to generate the strides.js file! See commit: https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/pull/24/commits/ce6e3accc1d9d4667c1b59b95dac1b720b05d540.

File: src/hooks/useCds/translate.js
Comment (mickohanlon23) [inline src/hooks/useCds/translate.js:318]:
```suggestion
  const mrn = patientData.Patient ? patientDataMap.Patient[0]?.identifier?.find(id => id.type?.text === 'MRN')?.value : null;
```
If I run the dashboard with the STRIDES data loaded, I get the following error:

Strides Mapping Error

Upon the dashboard's first load, the patientData array is empty, so line 310 is throwing an error. Something like what I suggested above should work, although you may be able to add handling higher in the file so that `mapStrideResult` is not called at all until all the patient data is loaded.

File: package.json
Comment (mickohanlon23) [inline package.json:105]: Do you want to add README comments about when a user would need to run the prestart and prebuild / prebuild-windows scripts? There are still some unresolved comments from my last review relating to the README, so you can probably address them all in one commit.

**Review by mickohanlon23 — CHANGES_REQUESTED** [review]
Nice! Very clever way to handle the strides.js file generation with the prebuild script. The testing works with babel, and all tests are passing, after merging in the updated pilot branch containing the EpisodeOfCare additions.

I was running into an error that I left a comment about. There were also a few unresolved comments from my last review around typos that should still be addressed. Other than that, nothing else major to address here.

Let's aim to merge this in by COB Thursday so that we can include these updates in the version of the Dashboard we move to the UMMC SUP environment.

File: package.json
Comment (yunwwang) [reply package.json:105]: Those pre scripts run automatically.

File: src/hooks/useCds/translate.js
Comment (mickohanlon23) [reply src/hooks/useCds/translate.js:318]: Nice! I see that this was fixed by adding empty array and null checks at the top of the `mapStrideResult` and `patientDataToHash`. I retested and this situation no longer throws this error in the browser.

File: package.json
Comment (mickohanlon23) [reply package.json:105]: Ahh, that's a nice feature, I hadn't realized that. This should be fine then, we'll just need to tell UMMC to run the script to convert the strides CSV file to JSON, which is already documented in the README.

**Review by mickohanlon23 — APPROVED** [review]
Great - Ready to test in the pilot SUP environment!

**Pull request merged**
